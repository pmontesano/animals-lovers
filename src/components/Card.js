import React from 'react';

const Card = ({ photo, alt, title, action, handleClick, id }) => {
  const handleButtonClick = (id) => (e) => {
    handleClick(id);
  };

  return (
    <div className="card">
      <div className="card-photo">
        <img className="mx-auto" src={photo} alt={alt} />
      </div>

      <div className="px-6 py-4">
        <div className={`${title ? 'font-bold text-xl mb-2' : 'skeleton-box'}`}>
          {title}
        </div>
      </div>
      <div className="px-6 py-4">
        {action && (
          <button
            className="btn btn-blue btn-blue:hover"
            onClick={handleButtonClick(id)}
          >
            {action}
          </button>
        )}
      </div>
    </div>
  );
};

export default Card;
