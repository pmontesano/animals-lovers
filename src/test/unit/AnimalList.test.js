import React from 'react';
import { create } from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import AnimalList from '../../components/AminalList';
import Card from '../../components/Card';

configure({ adapter: new Adapter() });

describe('AnimalsList', () => {
  it('should render correctly', () => {
    const props = {
      animals: [{ id: '1234', name: 'Bear', photo: 'bear.svg' }],
      handleClick: jest.fn,
    };

    const wrapper = create(<AnimalList {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render card component inside', () => {
    const props = {
      animals: [{ id: '1234', name: 'Bear', photo: 'bear.svg' }],
      handleClick: jest.fn,
    };

    const wrapper = shallow(<AnimalList {...props} />);

    expect(wrapper.find(Card)).toHaveLength(1);
  });
});
