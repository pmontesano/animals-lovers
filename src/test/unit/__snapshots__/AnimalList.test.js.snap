// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimalsList should render correctly 1`] = `
<div>
  <ul
    className="flex flex-wrap -mx-2"
  >
    <li
      className="list-animals w-full sm:w1/2 md:w-1/4 px-2 my-6"
    >
      <div
        className="card"
      >
        <div
          className="card-photo"
        >
          <img
            className="mx-auto"
            src="bear.svg"
          />
        </div>
        <div
          className="px-6 py-4"
        >
          <div
            className="font-bold text-xl mb-2"
          >
            Users for Bear
          </div>
        </div>
        <div
          className="px-6 py-4"
        >
          <button
            className="btn btn-blue btn-blue:hover"
            onClick={[Function]}
          >
            See details
          </button>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
